src/v/kafka/server/replicated_partition.h

```diff
@@ -49,10 +56,25 @@ class replicated_partition final : public kafka::partition_proxy::impl {
     }
 
     model::offset high_watermark() const final {
+        if (_partition->is_read_replica_mode_enabled()) {
+            if (_partition->cloud_data_available()) {
+                return _partition->last_cloud_offset() + model::offset(1);
+            } else {
+                return model::offset(1);
```

Comment:
Should probably return 0 as well. I'm not sure 100%, perhaps you can check what values are returned by totally empty partition and use them in case if `cloud_data_available` returned `false`.

Reply:
yes, it should be 0, fixed